### DISCUSSION TITLE
Add "customise Theme from template" to style settings

### DISCUSSION THREAD

**Discussion by Gewerd-Strauss** [chrisgrieser/shimmering-focus#202] [Q&A] [answered]
Hello,

I love this theme, and use it's darkmode almost 99% of the time. Very rarely I feel the need to switch to light-mode, because my eyes seem to become tired much faster with in lightmode settings. 

However, there are a few times I require it, and everytime I do so something very specific irks me - the inability to quickly copy a preset and tinker with it under "custom colors". 

Let me explain.
I use the `gamma`-color scheme basically exclusively in both dark- and lightmode.  However, in light mode it is a bit too cold for me, and same goes for default - and thus I will need to edit custom colors myself.
The issue is that there are a shitton of (amazing) features in this theme - and configuring that manually from scratch to make minor changes can take several hours. And that time I don't have unfortunately for fiddling with this in painstaking details. 

I would appreciate the ability to base the `custom colors` on any one of the three presets, and then go from there. I only need to change a minute detail, but getting all the other things done will take longer than I am willing to spend. 
Alternatively, as the data for the settings have to be stored _somewhere_, would it be possible to just copy-paste the settings for `Gamma` into `custom colors`, then start Obsidian to finish off the tinkering? Where would that data be stored/can it be edited by a normal person? A short look into the `Shimmering Focus.css`-file scares me :P

Thank you.  
Sincerely,
~Gw

**Comment by chrisgrieser** [comment, marked as answer]
Hi!

Unfortunately, that is not possible via CSS. Creating customizable colors alone is quite work-intensive when done in CSS, resulting in that "scary" code 😝 

However, with 0.16 will come **a lot** of changes. For one, I had to fully rewrite the theme due to all the changes in Obsidian, which unfortunately also forced me to focus on the essentials. That means the number of settings is considerably lower for now (Re-creating all features will take quite some time, I don't have for now.) On the other hand, 0.16 will make it far easier to customize any theme yourself with a CSS snippet, so you will only need a very rudimentary knowledge of CSS to customize colors, which might be helpful for you?

**Reply by Gewerd-Strauss** [reply]
Oh I didn't know that that was not doable, but thank you for the clarification. We'll see what 0.16 brings, I am not yet sure I will even update all-that soon as I have  a shitton of \*counts, then gulps\* **73** plugins installed, and a lot of them are integral to my vault. 

I could probably learn css to the point requried, but so far I could stay away from it. It doesn't look all too terrifying overall, but then again I know shit nothing regarding details on that :P But I will probably manage, and if not, chances are high I'll find someone who does and then I'll frankenstein it from there.

Although I _am_ curious what counts as "essentials" in a theme so incredibly expansive as SF.

**Reply by chrisgrieser** [reply]
The plugin's are mostly not affected, it's mainly themes and CSS snippets which have to be rewritten. Shimmering Focus already has a new version 3 which is compatible with Obsidian 0.16.

With Obsidian 0.16, it will be fairly easy to create your own color scheme, I have added [some instruction for that](https://github.com/chrisgrieser/shimmering-focus#create--share-your-own-color-scheme).

> Although I am curious what counts as "essentials" in a theme so incredibly expansive as SF.

Well, the essentials are all the options for hiding UI elements – at first and foremost, SF is a minimalistic theme, even though some people re-enable hidden elements 😝
